### PR TITLE
Avoid using `PyCollector` - inherit from `Module` instead

### PR DESCRIPTION
**Pull request by bluetech**
`PyCollector` is technically private in pytest and might change/break. The `DescribeBlock`s act like modules, so instead treat them as ones directly.

---

Hi, over in pytest we are considering whether to expose `PyCollector` or treat it as an internal implementation detail (ref: https://github.com/pytest-dev/pytest/pull/9264). My opinion is that it should be private. Looking at plugins which might be affected by this, I found one case - pytest-describe. I believe that pytest-describe can avoid `PyCollector` with the change proposed here.

BTW, this plugin is really ingenious - treating functions as an extra Python namespacing mechanism by treating them as modules, that's a cool concept :)

### CODE REVIEW

File: pytest_describe/plugin.py [hunk 1]
```diff
@@ -78,10 +89,6 @@ def collect(self):
     def _getobj(self):
         return self._importtestmodule()
 
-    def _makeid(self):
```
Comment (bluetech) [inline pytest_describe/plugin.py:81]: By passing nodeid directly, we don't need this. Also, newer versions of pytest no longer use `_makeid`.

**Review by jacebrowning — APPROVED** [review]
Thanks! This looks reasonable to me.

@Cito, we might want to make a beta release with this change.

File: pytest_describe/plugin.py (hunk 1 above)
Comment (Cito) [reply pytest_describe/plugin.py:81]: Right, this was removed [here](https://github.com/pytest-dev/pytest/pull/3291) in pytest 3.5, and we now require pytest 4.0.

**Comment by Cito** [comment]
Thanks for the heads-up and PR, @bluetech.

@jacebrowning Will check this with a larger real-world project, and I think will then just publish a 2.0.1 patch release. There is no pytest 7 alpha or beta yet, and our test matrix does not cover it, so we will need to check this again when pytest 7 comes out.

**Comment by Cito** [comment]
Version 2.01 is out now.

**Comment by bluetech** [comment]
Great, thanks!

**Pull request merged**